osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/test/java/org/opensmartgridplatform/adapter/protocol/dlms/domain/commands/alarm/SetAlarmNotificationsCommandExecutorTest.java

```diff
@@ -0,0 +1,112 @@
+import org.opensmartgridplatform.adapter.protocol.dlms.domain.entities.DlmsDevice;
+import org.opensmartgridplatform.adapter.protocol.dlms.domain.factories.DlmsConnectionManager;
+import org.opensmartgridplatform.dto.valueobjects.smartmetering.AlarmNotificationDto;
+import org.opensmartgridplatform.dto.valueobjects.smartmetering.AlarmNotificationsDto;
+import org.opensmartgridplatform.dto.valueobjects.smartmetering.AlarmTypeDto;
+import org.opensmartgridplatform.shared.exceptionhandling.OsgpException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.HashSet;
+import java.util.Arrays;
+import static org.junit.Assert.assertEquals;
+
+
+public class SetAlarmNotificationsCommandExecutorTest {
+    private DlmsDevice device;
+    private CommandExecutor<AlarmNotificationsDto, AccessResultCode> executor;
+    private List<SetParameter> setParametersReceived;
+    private DlmsConnectionManager connMgr;
+
+    @Before
+    public void setUp() {
+        this.setParametersReceived = new ArrayList<>();
+        this.device = new DlmsDevice("SuperAwesomeHeroicRockstarDevice");
+
+        final DlmsObjectConfigConfiguration dlmsObjectConfigConfiguration = new DlmsObjectConfigConfiguration();
+        final DlmsObjectConfigService dlmsObjectConfigService = new DlmsObjectConfigService
+                (new DlmsHelper(), dlmsObjectConfigConfiguration.getDlmsObjectConfigs()) {
+            @Override
+            public Optional<AttributeAddress> findAttributeAddress(
+                    final DlmsDevice device,
+                    final DlmsObjectType type,
+                    final Integer channel) {
+                return Optional.of(new AttributeAddress(40, "0.0.97.98.10.255", 2));
+            }
+        };
+        this.executor = new SetAlarmNotificationsCommandExecutor(dlmsObjectConfigService);
+
+        final DlmsConnectionStub conn = new DlmsConnectionStub() {
+            @Override
+            public AccessResultCode set(final SetParameter setParameter) {
+                setParametersReceived.add(setParameter);
+                return AccessResultCode.SUCCESS;
+            }
+        };
+
+        // Set the return value to 10 (0b1010):
+        // REPLACE_BATTERY enabled, AUXILIARY_EVENT enabled.
+        conn.addReturnValue(
+                new AttributeAddress(1, "0.0.97.98.10.255", 2),
+                DataObject.newInteger32Data(10));
+
+        this.connMgr = new DlmsConnectionManagerStub(conn);
+    }
+
+    @Test
+    public void testSetSettingThatIsAlreadySet() throws OsgpException {
+        // Setting notifications that are not different from what is on the meter already,
+        // should always be successful.
+        AccessResultCode res = this.execute(
+                new AlarmNotificationDto(AlarmTypeDto.REPLACE_BATTERY, true));
+        assertEquals(AccessResultCode.SUCCESS, res);
```

Comment:
in all unit tests, AssertJ language is used. Best to keep that aligned.

Reply:
Done.